addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -468,9 +470,19 @@ async function loadImageInfo(img, rpc, attachmentSrc = '') {
     // The "redirect" check is for when it is a redirect image attachment due to
     // an external URL upload.
     if (original && original.image_src && !/\/web\/image\/\d+-redirect\//.test(original.image_src)) {
+        if (img.dataset.originalSrc && !img.dataset.mimetypeBeforeConversion) {
+            // There is only the 'mimetypeBeforeConversion' attribute missing.
+            // For example, images that were uploaded on the website in version
+            // prior to saas-16.4 (when the 'mimetypeBeforeConversion' was
+            // introduced as a stable fix) and that did not go through
+            // `loadImageInfo` already.
+            img.dataset.mimetypeBeforeConversion = original.mimetype;
+            return;
+        }
```

Comment:
I don't think this is necessary: it does not hurt to "re-add" (it won't do anything really?) the items that are already there is it? Part of this explanation could be move next to the above `if ((img.dataset.originalSrc && img.dataset.mimetypeBeforeConversion) || !src) {`

Reply:
> I don't think this is necessary: it does not hurt to "re-add" (it won't do anything really?) the items that are already there is it? 

Actually I think it could lead to a bug if this is removed. Indeed, since [this commit](https://github.com/odoo/odoo/commit/0449fe85cb0e1d639a4e1aeba26e90906f79254d), uploaded images are converted to wepb. If a jpeg image was uploaded before this commit, its mimetype would be webp after the upload. However, the merge of this commit without `if (img.dataset.originalSrc && !img.dataset.mimetypeBeforeConversion)` would lead `loadImageInfo` to transform the mimetype of the image to jpeg (because of the `img.dataset.mimetype = original.mimetype`) and so, at the end of the process, the mimetype of the image would be jpeg and not webp as wanted.

Reply:
@loco-odoo I am not sure to understand: if that's the case, that would mean the get_image_info route cannot be trusted? :grimacing: Were you able to reproduce that issue? Reminder: I probably will want to not rely on those dataset.stuff thingy anymore in the future and always ask the route for image infos... if we cannot trust its result, it is to be fixed :)